### PR TITLE
Cleaned up indentation, markdown, spelling, and more

### PR DESCRIPTION
**Pull request by olehermanse**
- Added script to process markdown files
- Ran script to remove trailing whitespace
- Ran script to normalize newlines before EOF
- Ran script to replace non-ascii characters
- Ran script to clean up whitespace in code blocks
- Manually edited markdown files with warnings
- Fixed typo: promsie
- Corrected capitalization of CFEngine

### CODE REVIEW

File: reference/promise-types/files.markdown [hunk 1]
```diff
@@ -767,12 +766,12 @@ itself; in other words, a single destination directory. So `src/subdir/file` wil
...
 ```cf3
-     body copy_from mycopy(from,server)
-     {
-     source      => "$(from)";
-     servers     => { "$(server)" };
-     collapse_destination_dir => "true";
-     }
+body copy_from mycopy(from,server)
+{
+source      => "$(from)";
```
Comment (nickanderson) [inline reference/promise-types/files.markdown:771]: usually attributes are indented  but i can see they were not here and the fixup just fixed the whole block indention, so ok

File: web-ui/enterprise-reporting/reporting_ui.markdown [hunk 2]
```diff
@@ -13,12 +13,12 @@ If you are familiar with SQL syntax, you can input your query into the interface
...
-  SQL_QUERY="SELECT Hosts.HostName AS 'Host Name' FROM Hosts"
-  REPORT_TITLE="Example Report"
-  LINK="${HUB_URL}${API}$(echo ${SQL_QUERY} | base64)&title=$(/usr/bin/urlencode ${REPORT_TITLE})"
-  echo "${LINK}"
+HUB_URL="https://hub"
+API="/index.php/advancedreports/#/report/run?sql="
+SQL_QUERY="SELECT Hosts.HostName AS 'Host Name' FROM Hosts"
+REPORT_TITLE="Example Report"
+LINK="${HUB_URL}${API}$(echo ${SQL_QUERY} | base64)&title=$(/usr/bin/urlencode ${REPORT_TITLE})"
+echo "${LINK}"
```
Comment (nickanderson) [inline web-ui/enterprise-reporting/reporting_ui.markdown:21]: code block cleanup looked ok, except for the one cheat-sheet case where I think the indention was intended.

File: resources/additional-topics/modularity.markdown [hunk 3]
```diff
@@ -993,7 +993,7 @@ If we break the tower, by giving it an impossible promise to keep, e.g. changing
 the name of the directory in tier 3 to something that cannot be created2, then
 tier 3 will fail and the output looks like this:
 
-``console
+```console
```
Comment (nickanderson) [inline resources/additional-topics/modularity.markdown:996]: artisanally crafted code blocks look OK to me.

heh some good wonks in there, quadruple backtick

File: resources/faq/why-are-remote-agents-not-updating.markdown [hunk 4]
```diff
@@ -28,7 +28,7 @@ clients will see no need to scan for updates.
 
 * Check that the policy on in `$(sys.masterdir)` on the hub validates with
   `cf-promises`.
-* Check if `$(sys.inputdir)/cf_promsies_validated` differs from the
+* Check if `$(sys.inputdir)/cf_promises_validated` differs from the
```
Comment (nickanderson) [inline resources/faq/why-are-remote-agents-not-updating.markdown:31]: Did you git blame them? I bet they are all me!

File: web-ui/settings.markdown [hunk 5]
```diff
@@ -188,7 +188,7 @@ Mission portal can authenticate against an external directory.
 
 **Special Notes:**
 
-- LDAP API Url refers to the API cfengine uses internally for authentication.
+- LDAP API Url refers to the API CFEngine uses internally for authentication.
```
Comment (nickanderson) [inline web-ui/settings.markdown:191]: yay CFEngine, not `CF Engine`, or  `Cfengine`, or `CFengine`

**Review by nickanderson — APPROVED** [review]
Mostly good. But the ascii cleanup part left a lot of DEC CHAR 145s.

Few other things. Overall it's an improvement so id take it the way it is but would be nice if those ascii chars were tidied.

**Comment by nickanderson** [comment]
@cf-bottom jenkins preview build please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/132/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/132/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-132/output/_site/

File: markdowner.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
+# To run this on all markdown files do:
+# find . -name '*.markdown' -type f -exec python3 ./markdowner.py {} all \; | tee output.log
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply markdowner.py:3]: 1 file and N "operations" (codeblocks, ascii, trailing, eof, all).

File: reference/promise-types/files/edit_line/select_region-include.markdown [hunk 7]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply reference/promise-types/files/edit_line/select_region-include.markdown:1]: @nickanderson I guess we can delete this file in a follow-up PR?

File: web-ui/hub_administration/lookup-license-info.markdown [hunk 8]
```diff
@@ -36,4 +36,3 @@ License host key: SHA=2e5c7d9636c5644d023d71859f3296755f8d53d5d183af98efc1540655
 Expiration date:  3018-01-01
 Utilization:      20/50 (Approximate)
 ```
-
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply web-ui/hub_administration/lookup-license-info.markdown:39]: I suggest we delete those empty files in a followup commit (I can do it).

File: web-ui/settings.markdown [hunk 9]
```diff
@@ -227,18 +227,18 @@ Mission Portal's configuration can be exported and imported.
...
+**Special Notes:**
 
 - Admin role has all permissions by default.
 
 - Cf_remoteagent role has all permissions related to API by default.
 
 - Permissions granted by roles are additive, users with multiple roles are permitted to perform actions granted by each role.
 
-**Restore admin role permissions:** 
+**Restore admin role permissions:**
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply web-ui/settings.markdown:241]: Yep. Or I guess even better, add an action to run this script and ensure it edited 0 files. That would catch EOF, trailing, indentation, etc.

File: reference/promise-types/files/edit_line/select_region-include.markdown (hunk 7 above)
[the remark this answers is not in the document]
Comment (nickanderson) [reply reference/promise-types/files/edit_line/select_region-include.markdown:1]: Yep.

File: web-ui/settings.markdown (hunk 9 above)
[the remark this answers is not in the document]
Comment (nickanderson) [reply web-ui/settings.markdown:241]: +1

**Review by nickanderson — CHANGES_REQUESTED** [review]
Oh, fails the build:

```
13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute(cfengine)%]</p>
13:42:52 '
13:42:52 
13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute()%]</p>
13:42:52 '
13:42:52 
13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute(cfengine)%]</p>
```

File: reference/promise-types/files.markdown (hunk 1 above)
Comment (olehermanse) [reply reference/promise-types/files.markdown:771]: Yep, no autoformatting yet.

**Comment by nickanderson** [comment]
> Oh, fails the build:
> 
> ```
> 13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute(cfengine)%]</p>
> 13:42:52 '
> 13:42:52 
> 13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute()%]</p>
> 13:42:52 '
> 13:42:52 
> 13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute(cfengine)%]</p>
> ```

It must be broken in doc generator because it's not found in the expected heiracrchy of the site?

becasue that worked to render acl_inherit on 3.18 branch:
https://docs.cfengine.com/docs/3.18/reference-promise-types-files.html

File: examples/example-snippets/cfengine-administration.markdown [hunk 10]
```diff
@@ -11,7 +11,7 @@ tags: [Examples, CFEngine Administration]
 
 ## Ordering promises
 
-This counts to five by default. If we change ‘/bin/echo one’ to ‘/bin/echox one’, then the command will fail, causing us to skip five and go to six instead.
+This counts to five by default. If we change ‘/bin/echo one' to ‘/bin/echox one', then the command will fail, causing us to skip five and go to six instead.
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply examples/example-snippets/cfengine-administration.markdown]: Great catch, thanks! See the new commits I've added.

File: reference/language-concepts/loops.markdown [hunk 11]
```diff
@@ -77,8 +77,8 @@ The output looks something like this:
     2013-06-12T18:56:01+0200   notice: R: cf-execd is The executor, not executionist
 
 You see from this that, if we refer to a list variable using the scalar
-reference operator `$()`, CFEngine interprets this to mean “please iterate
-over all values of the list”. Thus, we have effectively a `foreach' loop,
+reference operator `$()`, CFEngine interprets this to mean "please iterate
+over all values of the list". Thus, we have effectively a `foreach' loop,
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply reference/language-concepts/loops.markdown:80]: Hmm, I didn't touch that one.

File: resources/faq/why-are-remote-agents-not-updating.markdown (hunk 4 above)
Comment (olehermanse) [reply resources/faq/why-are-remote-agents-not-updating.markdown:31]: Nope, not really important to blame :)

File: markdowner.py [hunk 12]
```diff
@@ -152,7 +152,7 @@ def perform_edits(content, flags, filename):
         content = replace_with_dict(content, replacements, filename)
 
     if flags.ascii or flags.all:
-        replacements = {"’": "'", "“": '"', "”": '"', "–": "-"}
+        replacements = {"‘":"'", "’": "'", "“": '"', "”": '"', "–": "-"}
```
Comment (nickanderson) [inline markdowner.py:155]: pesky

File: reference/language-concepts.markdown [hunk 13]
```diff
@@ -110,7 +110,7 @@ The CFEngine 3 language has a few simple rules:
 
     In each of these cases, the right hand side is a user choice.
 
-	CFEngine uses many `constraint expressions' as part of the body of a promise. These take the form: left-hand-side (cfengine word) ‘=>' right-hand-side (user defined data). This can take several forms:
+	CFEngine uses many `constraint expressions' as part of the body of a promise. These take the form: left-hand-side (cfengine word) '=>' right-hand-side (user defined data). This can take several forms:
```
Comment (nickanderson) [inline reference/language-concepts.markdown:113]: Good ascii fix but also, `cfengine` but not `CFEngine`?

File: resources/additional-topics/security.markdown [hunk 14]
```diff
@@ -249,7 +249,7 @@ server by using the trustkey directive. The trustkey directive in copy_from
 allows a client to decide whether to accept keys from a server. The CFEngine
 authentication model is based on the ssh scheme, however unlike ssh, CFEngine
 authentication is not interactive and the keys are generated by cf-key program
-instead of ‘ssh key-gen' program. Key acceptance is accomplished in CFEngine
+instead of 'ssh key-gen' program. Key acceptance is accomplished in CFEngine
```
Comment (nickanderson) [inline resources/additional-topics/security.markdown:252]: fixup dec 145 ascii looks good.

**Review by nickanderson — COMMENTED** [review]
Additional changes for ascii cleanup looks good. 

But, Will it blend?
will-it-blend

**Comment by nickanderson** [comment]
> But, Will it blend? 
will-it-blend

@cf-bottom jenkins, preview build please.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/134/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/134/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-134/output/_site/

**Comment by nickanderson** [comment]
> > But, Will it blend?
> > will-it-blend
> >       
> >         [
> >           
> >             will-it-blend
> >           
> >         ](https://user-images.githubusercontent.com/202896/182229936-a15a87be-6dae-4edf-8baf-71d345a894e8.gif)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://user-images.githubusercontent.com/202896/182229936-a15a87be-6dae-4edf-8baf-71d345a894e8.gif)
> 
> @cf-bottom jenkins, preview build please.

Nope, won't blend.

```
13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute(cfengine)%]</p>
13:42:52 '
13:42:52 
13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute()%]</p>
13:42:52 '
13:42:52 
13:42:52 Unexpanded macro in '/nt/documentation/reference/promise-types/files.markdown', html-line '<p>[%CFEngine_promise_attribute(cfengine)%]</p>
13:42:52 '
13:42:52 
13:42:52       Exception: (<class 'Exception'>, Exception('3 errors while processing HTML files'), <traceback object at 0x7f3e5bf4d440>)
```

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/135/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/135/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-135/output/_site/

File: reference/language-concepts.markdown (hunk 13 above)
Comment (olehermanse) [reply reference/language-concepts.markdown:113]: ACK; fixed.

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please :)

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/137/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/137/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-137/output/_site/

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/141/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/141/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-141/output/_site/

**Comment by olehermanse** [comment]
@cf-bottom Please build this again in Jenkins :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/143/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/143/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-143/output/_site/

**Comment by olehermanse** [comment]
@nickanderson This one is green and I guess good to go.

Issue was:

https://github.com/cfengine/documentation/pull/2817/commits/4ed5dd766d745e5659ca1409d3812cb880003a09

This should be fixed here:

https://github.com/cfengine/documentation-generator/pull/323

But that PR should wait until we've fixed other issues in documentation-generator

**Comment by olehermanse** [comment]
@cf-bottom Please build this in Jenkins one last time, after my changes in generator, just in case :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/147/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/147/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-147/output/_site/

**Review by nickanderson — APPROVED** [review]
tonight-we-test-in-prod-

File: web-ui/settings.markdown (hunk 9 above)
[the remark this answers is not in the document]
Comment (olehermanse) [reply web-ui/settings.markdown:241]: https://github.com/cfengine/documentation/pull/2824

File: web-ui/hub_administration/lookup-license-info.markdown (hunk 8 above)
[the remark this answers is not in the document]
Comment (olehermanse) [reply web-ui/hub_administration/lookup-license-info.markdown:39]: https://github.com/cfengine/documentation/pull/2827

File: reference/promise-types/files/edit_line/select_region-include.markdown (hunk 7 above)
[the remark this answers is not in the document]
Comment (olehermanse) [reply reference/promise-types/files/edit_line/select_region-include.markdown:1]: https://github.com/cfengine/documentation/pull/2827

**Pull request merged**